.buildkite/integration.pipeline.yml

```diff
@@ -97,13 +97,22 @@ steps:
         env:
           PACKAGES: "docker"
           PLATFORMS: "linux/amd64"
-        command: ".buildkite/scripts/steps/integration-package.sh"
+          CUSTOM_IMAGE_TAG: "git-${BUILDKITE_COMMIT:0:12}"
+          CI_ELASTIC_AGENT_DOCKER_IMAGE: "docker.elastic.co/beats-ci/elastic-agent-cloud"
+        command: |
+          source .buildkite/scripts/common.sh
+          .buildkite/scripts/steps/integration-package.sh
+          echo "~~~ Pushing cloud image"
+          mage cloud:push
         artifact_paths:
           - build/distributions/**
         agents:
           provider: "gcp"
           machineType: "n2-standard-8"
           diskSizeGb: 200
+        plugins:
+          - elastic/vault-docker-login#v0.5.2:
+              secret_path: 'kv/ci-shared/platform-ingest/elastic_docker_registry'
```

Comment:
Since you use this twice, I recommend using the anchor approach:
- https://buildkite.com/docs/pipelines/integrations/plugins/using#using-yaml-anchors-with-plugins

You can see how it looks also in the other pipelines.